dll/cpl/desk/screensaver.c

```diff
@@ -308,8 +304,8 @@ ScreenSaverConfig(HWND hwndDlg, PDATA pData)
             pData->PrevWindowPi.hThread = pData->PrevWindowPi.hProcess = NULL;
         }
 
-        if (WaitForSettingsDialog(hwndDlg, pi.hProcess))
-            SetScreenSaverPreviewBox(hwndDlg, pData);
+        WaitForSettingsDialog(hwndDlg, pi.hProcess);
+        SetScreenSaverPreviewBox(hwndDlg, pData);
```

Comment:
@Doug-Lyons I'm curious: What would happen here if you kept the WaitForSettingsDialog function to return the TRUE/FALSE booleans as it was done originally? Would it still cause some problems?
(For example, [I see I_Kill_Bugs saying this](https://jira.reactos.org/browse/CORE-18745?focusedId=137740&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-137740):
> I tried your desk.cpl and 3dText.scr in Windows XP and the configure dialog never opened. And if you keep clicking the settings button, it creates numerous windowspreview windows and never deletes them.

but I don't know whether this is some sort of side-effect due to the IsDialogMessage bug that was originally present in the function.)

_Note: the IsDialogMessage fix is correct, it needs to be kept._

Reply:
@HBelusca, I tested this and I was never able to get the function to return FALSE for some reason. In looking at the function, it appears that the only way to get this is that we receive a WM_QUIT message and I was never able to trigger this code path. Also, if we were able to return FALSE, the screen saver does not get re-triggered and would not run in the small display after returning from the 'settings' dialog. Just changing the returns function to VOID seemed to be the best path forward to me. Thanks.

Reply:
@HBelusca, I have not tested the new code in Windows XP, so I am not sure how it would work there. Thanks.

Reply:
> it appears that the only way to get this is that we receive a WM_QUIT message

Try pressing (and maintaining) alt-tab down so as to continuously switch windows; this may trigger the screensaver program to exit :thinking: Or, insert an artificial `TerminateProcess(pi.hProcess, 0);` call _**just before**_  that `WaitForSettingsDialog` call?

Reply:
@HBelusca, I have done more testing with the alt-tab and I was not able to ever see the WM_QUIT message being hit.
Also, I was not able to ever see a return with a value of "FALSE" from this function.
I did this testing with the corrected if statement adding the NOT into ```!IsDialogMessage``` statement.

Reply:
Thanks @Doug-Lyons , then let's proceed with this code simplification.